src/f32/vec2.rs

```diff
@@ -401,6 +401,36 @@ impl Vec2 {
             angle
         }
     }
+
+    /// Returns a `Vec2` with a length no less than `min` and no more than `max`
+    pub fn clamp_length(&self, min: f32, max: f32) -> Self {
+        let length_sq = self.length_squared();
+        if length_sq < min * min {
+            self.normalize() * min
```

Comment:
normalize will recalculate length squared again so it would be better to reuse the `length_sq` value that's already been calculated, e.g. replace normalize with something like `self * (length_sq.sqrt().recip() * min)`.

The normalize function is doing `self * self.length_squared().sqrt().recip()`

Reply:
Realized I missed the recip when I mentioned this on the issue.

Reply:
Ah, so that's what was going on with your example values. All I knew was the value was way off. I'll update this PR tomorrow, so at least it's a valid reference for the eventual implementation.

Reply:
Apologies, real life intervened so I wasn't able to turn this around when I had intended.

I've just pushed the optimization updates. 👍🏼  How is the macro-ification coming? Is it at a good state where I should look at porting this to a macro?